python_modules/libraries/dagster-components/dagster_components/utils.py

```diff
@@ -143,3 +155,77 @@ def load_module_from_path(module_name, path) -> ModuleType:
     assert spec.loader, "Must have a loader"
     spec.loader.exec_module(module)
     return module
+
+
+# ########################
+# ##### TOML MANIPULATION
+# ########################
+
+
+@contextmanager
+def modify_toml(path: Path) -> Iterator[tomlkit.TOMLDocument]:
+    with open(path) as f:
+        toml = tomlkit.parse(f.read())
+    yield toml
+    with open(path, "w") as f:
+        f.write(tomlkit.dumps(toml))
+
+
+def get_toml_value(doc: tomlkit.TOMLDocument, path: Iterable[str], expected_type: type[T]) -> T:
```

Comment:
this seems more like validate_toml_value? or get_and_validate_toml_value?

Reply:
It's primary function is to retrieve values from a toml document. The `expected_type` argument is just to make it a typed getter. Prefer to leave the name as is to match impl in `dagster_dg`.